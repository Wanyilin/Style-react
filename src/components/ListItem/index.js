import React from 'react';
import { convertCamel, formatter } from 'src/utils/formatter';

import './index.scss';

const ListItem = ({
	type,
	value,
	name,
}) => {
	const title = convertCamel(name);
	const content = formatter({ type, value, name });
	return (
		<div className="list-item">
			<span className="item-title">{title}</span>
			<span className="item-content">{content}</span>
		</div>
	)
};

export default ListItem;
