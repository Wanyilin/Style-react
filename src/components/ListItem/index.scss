@import 'src/theme.scss';

.list-item {
	display: flex;
	justify-content: space-between;
	align-items: center;
	padding-bottom: $ListItemMargin;

	&:last-child {
		padding-bottom: 0;
	}

	.item-title {
		@include FontStyle($ItemTitleSize, $ItemTitleWeight, $ItemTitleColor);
		letter-spacing: 0.021em;
	}
	.item-content {
		@include FontStyle($ItemContentSize, $ItemContentWeight, $MainFontColor);
		letter-spacing: 0.01em;
	}
}