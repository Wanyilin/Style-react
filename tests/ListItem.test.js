import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import ListItem from '../src/components/ListItem';

describe('Testt ListItem', () => {
   test('Should match snapshot', () => {
    const component = renderer.create(
			<ListItem type="type" value="value" name="name" />
		);
		let tree = component.toJSON();
  	expect(tree).toMatchSnapshot();
  });
  test('Should display context as expect', () => {
		const { getByText } = render(
			<ListItem type="type" value="value" name="name" />
		);
		expect(getByText('Name')).toBeInTheDocument();
		expect(getByText('Value')).toBeInTheDocument();
  })
});
