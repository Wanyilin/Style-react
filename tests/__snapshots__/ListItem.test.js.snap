// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Should match snapshot 1`] = `
<div
  className="list-item"
>
  <span
    className="item-title"
  >
    Name
  </span>
  <span
    className="item-content"
  >
    Value
  </span>
</div>
`;

exports[`Testt ListItem Should match snapshot 1`] = `
<div
  className="list-item"
>
  <span
    className="item-title"
  >
    Name
  </span>
  <span
    className="item-content"
  >
    Value
  </span>
</div>
`;
